File: src/main/java/com/smarthome/shcartservice/repo/CartRepository.java
package com.smarthome.shcartservice.repo;

import com.smarthome.shcartservice.entity.Cart;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface CartRepository extends JpaRepository<Cart, Long> {

    Optional<Cart> findCartByUserId(Long userId);

    void deleteCartByUserId(Long userId);

}

File: src/main/java/com/smarthome/shcartservice/repo/CartRepository.kt
package com.smarthome.shcartservice.repo

import com.smarthome.shcartservice.entity.Cart
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface CartRepository : JpaRepository<Cart, Long> {
    fun findCartByUserId(userId: Long): Optional<Cart>
}
